spyder/plugins/completion/languageserver/client.py

```diff
@@ -288,22 +286,23 @@ def patched_cleanup():
         logger.debug('LSP {} client started!'.format(self.language))
 
     def stop(self):
+        """Stop transport and server."""
         logger.info('Stopping {} client...'.format(self.language))
         if self.notifier is not None:
             self.notifier.activated.disconnect(self.on_msg_received)
             self.notifier.setEnabled(False)
             self.notifier = None
-        if self.transport_client is not None:
-            self.transport_client.kill()
+        if self.transport is not None:
+            self.transport.kill()
         self.context.destroy()
-        if self.lsp_server is not None:
-            self.lsp_server.kill()
+        if self.server is not None:
+            self.server.kill()
 
     def is_transport_alive(self):
         """Detect if transport layer is alive."""
         alive = True
-        if self.transport_client is not None:
-            if self.transport_client.poll() is not None:
+        if self.transport is not None:
+            if self.transport.state() == QProcess.NotRunning:
                 alive = False
```

Comment:
How about:

```python
        alive = False
        if self.transport_client is not None:
            alive = self.transport.state() != QProcess.NotRunning:
```

Why is the default value True for alive. If there is no transport, it will return True which feels weird.

Reply:
I also refactored this.